torch_em/data/datasets/pannuke.py

```diff
@@ -0,0 +1,149 @@
+import os
+import h5py
+import vigra
+import shutil
+import numpy as np
+from glob import glob
+
+import torch_em
+from torch_em.util.debug import check_loader
+from torch_em.data.datasets import util
+
+
+# PanNuke Dataset - https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke
+URLS = {
+    "fold_1": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_1.zip",
+    "fold_2": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_2.zip",
+    "fold_3": "https://warwick.ac.uk/fac/cross_fac/tia/data/pannuke/fold_3.zip"
+}
+
+
+# TODO
+CHECKSUM = None
+
+
+def _download_pannuke_dataset(path, download, folds):
+    os.makedirs(path, exist_ok=True)
+
+    checksum = CHECKSUM
+
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        util.download_source(os.path.join(path, f"{tmp_fold}.zip"), URLS[f"{tmp_fold}"], download, checksum)
+
+        print(f"Unzipping the PanNuke dataset in {tmp_fold} directories...")
+        util.unzip(os.path.join(path, f"{tmp_fold}.zip"), os.path.join(path, f"{tmp_fold}"), True)
+
+
+def _convert_to_hdf5(path, folds):
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        print(f"Converting the {tmp_fold} into h5 file format...")
+        tmp_name = tmp_fold.split("_")[0] + tmp_fold.split("_")[1]  # name of a particular sub-directory (per fold)
+        with h5py.File(os.path.join(path, f"pannuke_{tmp_fold}.h5"), "w") as f:
+            img_path = glob(os.path.join(path, tmp_fold, "*", "images", tmp_name, "images.npy"))[0]
+            gt_path = glob(os.path.join(path, tmp_fold, "*", "masks", tmp_name, "masks.npy"))[0]
+
+            f.create_dataset("images", data=np.load(img_path).transpose(3, 0, 1, 2))
+            f.create_dataset("masks", data=np.load(gt_path).transpose(3, 0, 1, 2))
```

Comment:
Use `compression="gzip"` when storing hdf5 datasets. This will save a lot of memory, especially for the segmentation data.